### PR TITLE
Revert #508

### PR DESCRIPTION
**Pull request by etspaceman**
This causes an issue in GHE runners that may not use the default java version of 11.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -484,11 +484,25 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
+        java: [temurin@11]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout current branch (fast)
         uses: actions/checkout@v3
 
+      - name: Setup Java (temurin@17)
```
Comment (armanbilge) [inline .github/workflows/ci.yml:493]: Actually there was a mistake here before I realized I could delete it. The matrix specifies 11 but the generated steps set up 17 ...

File: ci/src/main/scala/org/typelevel/sbt/TypelevelCiPlugin.scala [hunk 2]
```diff
@@ -169,13 +169,13 @@ object TypelevelCiPlugin extends AutoPlugin {
             "validate-steward",
             "Validate Steward Config",
             WorkflowStep.Checkout ::
+              WorkflowStep.SetupJava(List(JavaSpec.temurin("17"))) :::
```
Comment (armanbilge) [inline ci/src/main/scala/org/typelevel/sbt/TypelevelCiPlugin.scala:172]: ... probably easiest to fix that by just setting this to 11.

Comment (etspaceman) [reply ci/src/main/scala/org/typelevel/sbt/TypelevelCiPlugin.scala]: Updated

File: .github/workflows/ci.yml (hunk 1 above)
Comment (etspaceman) [reply .github/workflows/ci.yml]: Updated

**Review by armanbilge — APPROVED** [review]
Thanks, annoying, but more robust 👍

**Pull request merged**